### DISCUSSION TITLE
Time dependent Material Objects

### DISCUSSION THREAD

**Discussion by makeclean** [idaholab/moose#20760] [Q&A General] [answered]
Hi All

Looking at the documentation it seems that PiecewiseLinearInterpolationMaterials are aware of the standard variables, presumably x,y and z c.f. https://mooseframework.inl.gov/moose/source/materials/ADPiecewiseLinearInterpolationMaterial.html
```
[Materials]
  [./mat]
    type = PiecewiseLinearInterpolationMaterial
    property = matprop
    variable = x
    x = '0 1'
    y = '0 10'
    outputs = all
  [../]
[]
```
When I try to u 't' as in a regular function for time, MOOSE complains that it can't find the variable. I've hacked it to do something like
```
[AuxVariables]
  [time]
  []
[]
[AuxKernels]
  [time]
    type = FunctionAux
    variable = time
    function = 't'
    execute_on = timestep_begin
  []
```
Is that the best/right way to allow time dependent material objects?

**Comment by GiudGiud** [comment, marked as answer]
Hello

It depends how you want to provide this dependence. If you have a functional expression, I like to use ParsedMaterials, which can use t in the expression.

This material you are using uses a variable as its abscissa. This is a little overkill for time. 
Another option is to use a function:
with https://mooseframework.inl.gov/moose/source/materials/GenericFunctionMaterial.html 

This is similar to ParsedMaterials for simple time dependence, but the spatial dependence may be more complex with this form. 

Guillaume

**Comment by makeclean** [comment]
I guess I was also trying to highlight, given my experience with functions, why 't' didn't just work - for consistency I would have expected it to.

However, that gets me moving :) Thanks!
